File: Data Structures & Algorithms/binary-search/submission-0.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l , h = 0, len(nums) - 1
        mid = 0
        while l <= h:
            mid = int(l + (h - l) / 2)
            if nums[mid] == target:
                return mid
            elif nums[mid] > target:
                h = mid - 1
            elif nums[mid] < target:
                l = mid + 1
        return -1

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
      int cnt = 1;
      vector<vector<int>>maxi(nums.size()+1);
      vector<int>aa;
      sort(nums.begin(),nums.end());
      for(int i=0;i<nums.size();i++){
        if(nums[i]==nums[i+1]){
            cnt++;
        }
        else{
            maxi[cnt].push_back(nums[i]);
            cnt = 1;
        }
      }
      int n = maxi.size();
      for(int i=n-1; i>=1; i--){
       for (int num : maxi[i]) {
                aa.push_back(num);
                if (aa.size() == k) return aa; 
            }
      }
      return aa;
    }
};